nengo_gui/components/component.py

```diff
@@ -31,9 +35,9 @@ class Template(object):
     def __init__(self, *args, **kwargs):
         self.args = args
         self.kwargs = kwargs
-    def create(self, vizsim):
-        uid = vizsim.viz.get_uid(self)
-        c = self.cls(vizsim, vizsim.viz.config[self], uid,
+    def create(self, sim):
+        uid = '_uid_%d' % id(self)
+        c = self.cls(sim, sim.config[self], uid,
```

Comment:
I think, `Template` should at least declare `cls` with some default value (`None`, `object`, or whatever) just to make clear that this attribute is supposed to exist (and to be set by subclasses). Or is this not possible because it is a class attribute?